src/Sylius/Bundle/AdminBundle/Notification/GusNotificationProvider.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Notification;
+
+use Psr\Clock\ClockInterface;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Sylius\Bundle\CoreBundle\SyliusCoreBundle;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+final class GusNotificationProvider implements NotificationProviderInterface
+{
+    public const LATEST_SYLIUS_VERSION_CACHE_KEY = 'latest_sylius_version';
+
+    public function __construct(
+        private readonly ClientInterface $client,
+        private readonly RequestStack $requestStack,
+        private readonly RequestFactoryInterface $requestFactory,
+        private readonly StreamFactoryInterface $streamFactory,
+        private readonly CacheInterface $cache,
+        private readonly ClockInterface $clock,
+        private readonly string $hubUri,
+        private readonly string $environment,
+        private readonly int $checkFrequency,
+    ) {
+    }
+
+    public function getNotifications(): array
+    {
+        $metadata = $this->cache instanceof ItemInterface ? $this->cache->getMetadata() : [];
+        $metadata[ItemInterface::METADATA_EXPIRY] = $this->clock->now()->modify(sprintf('+%d minutes', $this->checkFrequency))->getTimestamp();
+
+        $latestVersion = $this->cache->get(self::LATEST_SYLIUS_VERSION_CACHE_KEY, function () {
+            return $this->getLatestVersion();
+        });
+
+        if (
+            $latestVersion === null ||
+            $latestVersion === SyliusCoreBundle::VERSION
```

Comment:
We'll probably need some better comparison here, no clue how the version from hub is structured 🤷

Reply:
For now, I would be for leaving it as it is, because it covers this functionality in the same way as it was in Sylius 1.x. But of course, we need to improve it in the future 🖖🏻